### PR TITLE
feat: add number input component

### PR DESCRIPTION
**Pull request by gabriel-piltzer-trimble**
### :page_facing_up: Summary of Changes

This PR adds the `modus-wc-number-input` component. It uses DaisyUI and Tailwind CSS.

### :thought_balloon: Type of Change

- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [ ] Configuration change (modifies scaling or properties of existing infrastructure & services)
- [ ] Documentation change
- [ ] Other

### :clipboard: Test Plan

Added unit tests

### :white_check_mark: Self Code Review Checklist

PR authors and reviewers, please verify that all of these items have been completed.

- [x] My code is clean and readable
- [x] I followed standard conventions
- [x] Component code is WCAG 2.2 compliant (if applicable)
- [ ] I have made theme files changes (if applicable)
- [ ] I have made documentation changes (if applicable)
- [x] I have made Storybook changes including usage documentation (if applicable)
- [x] I have tested the component thoroughly in Storybook including RTL rendering (if applicable)

### :link: Azure DevOps Work Item

[AB#576517](https://dev.azure.com/ViewpointVSO/74194628-0b6e-4b03-95a4-2d06c069dcbe/_workitems/edit/576517)

### CODE REVIEW

**Review by Kai-Richardson — COMMENTED** [review]
`max` and `min` do not seem to work properly. The up and down arrows prevent me from going past them, but i can put in an illegal number just fine.

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+export const Template: Story = {
+  render: (args) => html`
+    <modus-wc-number-input
+      aria-describedby=${ifDefined(args['aria-describedby'])}
+      aria-label=${args['aria-label']}
+      auto-complete=${ifDefined(args['auto-complete'])}
+      ?auto-focus=${args['auto-focus']}
+      ?bordered=${args.bordered}
+      custom-class=${args['custom-class']}
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]:
```suggestion
      custom-class=${ifDefined(args['custom-class'])}
```

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      ?disabled=${args.disabled}
+      input-aria-invalid=${ifDefined(args['input-aria-invalid'])}
+      input-dir=${ifDefined(args['input-dir'])}
+      input-id=${ifDefined(args['input-id'])}
+      input-mode=${args['input-mode']}
+      input-tab-index=${ifDefined(args['input-tab-index'])}
+      max=${ifDefined(args.max)}
+      min=${ifDefined(args.min)}
+      name=${ifDefined(args.name)}
+      placeholder=${args.placeholder}
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]:
```suggestion
      placeholder=${ifDefined(args.placeholder)}
```

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      ?read-only=${args['read-only']}
+      ?required=${args.required}
+      size=${args.size}
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]:
```suggestion
      size=${ifDefined(args.size)}
```

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      type=${args.type}
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]:
```suggestion
      type=${ifDefined(args.type)}
```

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts:1]: Just some minor issues with the template story

**Review by stuart-rimel — COMMENTED** [review]
Noticing from the storybook that the arrows on the scrollbar are cut off halfway. Is this a right padding issue?

chrome_kNTKa7KIoV

Also found some minor storybook issues

**Comment by gabriel-piltzer-trimble** [comment]
> Noticing from the storybook that the arrows on the scrollbar are cut off halfway. Is this a right padding issue?
> 
> chrome_kNTKa7KIoV
> 
> Also found some minor storybook issues

Look decent on my end...
image

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts (hunk 1 above)
Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Fixed

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts (hunk 2 above)
Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Fixed

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts (hunk 3 above)
Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Fixed

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts (hunk 4 above)
Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Fixed

File: src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      input-mode=${args['input-mode']}
+      input-tab-index=${ifDefined(args['input-tab-index'])}
+      max=${ifDefined(args.max)}
+      min=${ifDefined(args.min)}
+      name=${ifDefined(args.name)}
+      placeholder=${ifDefined(args.placeholder)}
+      ?read-only=${args['read-only']}
+      ?required=${args.required}
+      size=${ifDefined(args.size)}
+      step=${args.step}
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Uno mas

```suggestion
      step=${ifDefined(args.step)}
```

Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-number-input/modus-wc-number-input.stories.ts]: Fixed, whoops!

**Review by stuart-rimel — APPROVED** [review]

**Pull request merged**